appengine/flexible/endpoints/README.md

```diff
@@ -93,3 +93,71 @@ To use the client ID for authentication:
 Now you can use the client ID to make requests to the API:
 
     $ python clients/google-id-token-client.py https://YOUR-PROJECT-ID.appspot.com YOUR-API-KEY /path/to/client-id.json
+
+### Using the App Engine default service account client (no key file needed)
+
+The App Engine default service account client demonstrates how to use the Google App Engine default service account to authenticate to endpoints.
+To use the client, you need to create a Google Cloud project as your client project.
+
+To use the App Engine default service account for authentication:
+
+1. Update the `gae_default_service_account`'s `x-issuer` and `x-jwks_uri` in `swagger.yaml` with your client project ID.
+2. Redeploy your server application.
+3. Update clients/service_to_service_gae_default/main.py, replace `YOUR-CLIENT-PROJECT-ID` and `YOUR-SERVER-PROJECT-ID` with your client project ID and your server project ID.
+4. Upload your application to Google App Engine by invoking the following command under clients/service_to_service_gae_default directory.
+This opens a browser window for you to sign in using your Google account. You'll be providing the project ID as the argument for -A. Use
+the -V argument to specify a version name. Additional information on how to deploy an app to Google Cloud App Engine can be found [here](https://cloud.google.com/appengine/docs/python/quickstart).
+
+   appcfg.py -A <YOUR-CLIENT-PROJECT-ID> -V v1 update .
+
+Your client app is now deployed at https://<YOUR-CLIENT-PROJECT-ID>.appspot.com. When you access https://<YOUR-CLIENT-PROJECT-ID>.appspot.com, your client calls your server project API using
+the client's service account.
+
+### Using the service account client (no key file needed)
```

Comment:
According to Mingliang, the main complain from customers is that they need to manage key rotation. In this case, the key is not managed by the users but the cloud itself, which solves the issue.